rfcs/0011-project-structure.md

```diff
@@ -0,0 +1,116 @@
+
+Moreover, the latest developments in the [VS Code Extension](https://github.com/lf-lang/vscode-lingua-franca) include the introduction of a [Lingua Franca Package Explorer](https://github.com/lf-lang/lingo/wiki/Lingua-Franca-%E2%80%90-VS-Code-Extension). This tool allows users to easily search for and import pre-existing reactors into their current projects. These libraries can originate from two sources: those crafted by the programmer within their local workspace (Local Libraries) and those downloaded using the Lingo Package Manager (Lingo Libraries). This feature significantly enhances productivity and encourages the use of well-tested, modular code components. However, to effectively manage and list these libraries of reactors, it's essential to establish a designated access point within the project structure. This is typically achieved through a centralized directory, ensuring developers can seamlessly access and manage these libraries. This structured approach prevents the non-systematic scattering of libraries throughout the project, promoting organization and ease of maintenance.
+
+Maintaining a fixed structure is crucial because it allows developers to quickly identify where specific components, configurations, and resources are located, which in turn reduces the time spent searching for files and fosters efficient collaboration among team members. Therefore, to achieve these objective, this RFC proposes implementing a clear and standardized project structure for developers to follow. This standardized approach ensures projects are consistently organized, making them easier to comprehend, maintain, and share. Moreover, this structure facilitates the integration of external libraries and tools, thereby fostering a more collaborative development environment.
+
+# Proposed Implementation
+[proposed-implementation]: #proposed-implementation
+
+The current structure of a LF project is organized as follows:
+
+```shell
+├── root
+│   ├── bin/       # Binary executables
+│   ├── include/   # Header files
+│   ├── src/       # LF programs
+│   ├── src-gen/   # Generated code for local LF programs
+│   └── fed-gen/   # Generated code for federated programs (instead of src-gen)
+```
+
+To enhance the project structure, the following additions are proposed:
+
+- **lib/**: Directory for storing reusable reactors.
+- **Lingo.toml**: Configuration file crucial for project sharing, storing Lingo configuration details including external LF projects to include (refer to [Lingo Documentation](https://github.com/lf-lang/lingo?tab=readme-ov-file#the-toml-based-package-configurations) for further details).
```

Comment:
Perhaps we can also point out that the structure of `Lingo.toml` will be discussed in another RFC.

Reply:
Sure, I’ll add this information in the document